Automattic/wp-calypso
# State: Memoize some inline help selectors

#### Changes proposed in this Pull Request

While profiling Calypso, I noticed there are several components that needlessly re-render when any state change occurs.  One of them is `HelpSearchResults`. This happens because the relevant selectors essentially generate a new object every time, even if the state is the same.

This PR improves the responsible inline help search results selectors, so it would use the same object when the same data is used, reducing the number of re-renders of the My Home page.

#### Testing instructions

* Make sure you have React DevTools enabled. 
* Make sure you have the "Highlight updates when components render." option enabled.
* Load `/home/:site` where `:site` is one of your WP.com sites.
* Play with expanding and collapsing the Quick Links accordion with some delay between those actions.
* Verify the inline help search results component at the bottom doesn't re-render.
